src/wattsi.pas

```diff
@@ -904,6 +904,59 @@    TCrossReferences = record
       end;
    end;
 
+   procedure InsertMDNAnnotationForElement(const Element: TElement);
+   var
+      Candidate: TNode;
+      ID: UTF8String;
+      TargetAncestor, MDNBox: TElement;
+   begin
+      if (not(Element.HasAttribute('id'))) then
+         exit;
+      ID := Element.GetAttribute('id').AsString;
+      if (not(MDNJSONData[ID] is TJSONArray)) then
+         // No MDN article has a link to this ID.
+         exit;
+
+      MDNBox := E(eAside, ['class', 'mdn before wrapped']);
+
+      if (Element.HasProperties(propHeading)) then
+      begin
+         // If this is a heading, we insert the annotation after the element
+         // being annotated. When generating multipage output, inserting the
+         // annotation after the heading ensures that it ends up in the right
+         // file. It can otherwise incorrectly end up in the previous split.
+         MDNBox := E(eAside, ['class', 'mdn wrapped']);
+         if ((Element.NextSibling is TElement)
+               and (TElement(Element.NextSibling)
+                  .GetAttribute('class').AsString = 'mdn wrapped')) then
```

Comment:
What if the class is `'mdn before wrapped'`?

Reply:
> What if the class is `'mdn before wrapped'`?

If we have an `'mdn wrapped'` anno, we don’t actually want to merge it into any `'mdn before wrapped'` anno which might be at point where we’re evaluating this. Doing that will cause it to end up losing presentational alignment with the element it’s annotating.